nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -219,7 +219,10 @@ def main():
 
     gens = get_generations()
     for profile in get_profiles():
-        gens += get_generations(profile)
+        if set(profile) & set(r'><:"/\|?*'):
+            print("ignoring profile '{}' as it contains characters that are not valid for filenames on vfat filesystems".format(profile), file=sys.stderr)
```

Comment:
Wouldn't it be easier to wrap the `write_entry` (or the `open`) in a `try-except OSError` block? I guess there are more unicode chars out there that can be dangerous...
And maybe something like "ignoring profile '{}' _in the list of boot entries_"? Otherwise the message from `nixos-rebuild` might be a bit confusing, because the profile itself actually exists.

Reply:
Yes, you are right.  I have forgotten that this is Python and that we ask here for forgiveness and not for permission ;)